flycheck.el

```diff
@@ -3610,6 +3650,55 @@ list."
       ;; highlighting
       (flycheck-error-list-highlight-errors preserve-pos))))
 
+(defvar flycheck-error-list-current-filter nil
+  "The minimum level of errors to display in the error list.
+
+If set to an error level, only displays errors whose error level
+is at least as severe as this one in the error list.  If nil,
+display all errors.")
+
+(defun flycheck-error-list-mode-line-filter-indicator ()
+  "Create a string representing the current error list filter."
+  (if flycheck-error-list-current-filter
+      (format " [>= %s]" (symbol-name flycheck-error-list-current-filter))
+    ""))
+
+(defun flycheck-error-list-query-level (prompt)
+  "Reads an error level from the user.
+
+Only offers level for which errors currently exist."
+  (-when-let* ((error-levels (mapcar #'flycheck-error-level
+                                     (flycheck-error-list-current-errors)))
+               (uniq-error-levels (-uniq error-levels))
```

Comment:
I think we should always offer all levels.  A user might want to “proactively” filter warnings and infos, for instance, if they are about to make a larger refactoring, and fix up compilation errors first, before dealing with the new warnings.